api/app/helpers/spree/api/api_helpers.rb

```diff
@@ -177,15 +177,16 @@ def required_fields_for(model)
       ]
 
       def variant_attributes
-        if @current_user_roles && @current_user_roles.include?("admin")
+        if @current_user_roles&.include?("admin")
           @@variant_attributes + [:cost_price]
         else
           @@variant_attributes
         end
       end
 
       def total_on_hand_for(object)
-        object.total_on_hand.finite? ? object.total_on_hand : nil
+        total_on_hand = object.total_on_hand
+        total_on_hand if total_on_hand.finite?
```

Comment:
I'm not sure about my feelings on this one. This removes clarity in reading the code since the reader has to think that if the last if statement on a method doesn't result in any code being run, it returns nil, which is not explicitly stated. Is this change a Rubocop suggestion?

Reply:
I agree, this doesn't read as cleanly imo

Reply:
This was a code smell from reek, but if this is not preferable we don't need to make the change.

I have changed the PR just to swap that nil check with the safe navigation operator.